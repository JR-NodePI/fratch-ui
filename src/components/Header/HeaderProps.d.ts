import { ICON_POSITION } from './HeaderConstants';

export type HeaderProps = {
  title?: string;
  iconSrc?: string;
  iconPosition?: (typeof ICON_POSITION)[keyof typeof ICON_POSITION];
};
