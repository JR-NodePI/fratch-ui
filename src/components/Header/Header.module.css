.header {
  z-index: 2;
  -webkit-app-region: drag;
  background-color: var(--ft-header-bg-color);
  box-shadow: inset 0 0 10px var(--ft-primary-default-bg-color);
  color: var(--ft-primary-default-color);
  text-shadow:
    -1px 1px 3px var(--ft-secondary-default-bg-color),
    1px 1px 2px var(--ft-primary-hover-bg-color),
    1px -1px 1px var(--ft-color-light);
  text-transform: uppercase;
  font-size: var(--ft-font-size);
  margin: 0;
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
}
.header .icon {
  position: absolute;
  width: 24px;
  margin: 0 6px;
  top: 50%;
  transform: translateY(-50%);
}
.header .icon.left {
  left: 0;
}
.header .icon.right {
  right: 0;
}
.header h1 {
  text-align: center;
  line-height: var(--ft-header-height);
  font-size: var(--ft-font-size);
  margin: 0;
  font-weight: 300;
  letter-spacing: 0.3em;
}
