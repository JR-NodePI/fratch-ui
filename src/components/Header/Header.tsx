import { c } from '../../helpers/classNameHelpers';
import styles from './Header.module.css';
import logo from '../../assets/logo.png';

function Header({
  title,
  iconSrc = logo,
}: {
  title: string;
  iconSrc?: string;
}): JSX.Element {
  return (
    <div className={c(styles.header)}>
      <h1>
        <img className={c(styles.icon)} src={iconSrc} alt={title} />
        {title}
      </h1>
    </div>
  );
}

export default Header;
