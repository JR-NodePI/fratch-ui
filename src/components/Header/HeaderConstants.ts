export const ICON_POSITION = Object.freeze({
  LEFT: 'left',
  RIGHT: 'right',
} as const);
