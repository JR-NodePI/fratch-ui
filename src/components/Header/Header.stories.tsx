import type { Meta, StoryObj } from '@storybook/react';

import imgLogo from '../../assets/logo.png';
import Header from './Header';

const meta = {
  title: 'Components/Header',
  component: Header,
} satisfies Meta<typeof Header>;

export default meta;

type Story = StoryObj<typeof Header>;

export const _Header: Story = {
  args: {
    title: 'This is the title  ',
    imgLogo,
  },
};
