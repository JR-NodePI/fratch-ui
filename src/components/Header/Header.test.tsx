import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import Header from './Header';
import { type HeaderProps } from './HeaderProps';

describe('Header', () => {
  const setup = (props: HeaderProps = {}): ReturnType<typeof render> =>
    render(<Header {...props} />);

  it('should render properly', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should render with iconSrc', () => {
    const { container } = setup({ iconSrc: 'icon.png' });
    expect(container).toMatchSnapshot();
  });

  it('should render with title', () => {
    const { container } = setup({ title: 'mock title' });
    expect(container).toMatchSnapshot();
  });

  it('should render with iconPosition right', () => {
    const { container } = setup({ iconSrc: 'icon.png', iconPosition: 'right' });
    expect(container).toMatchSnapshot();
  });
});
