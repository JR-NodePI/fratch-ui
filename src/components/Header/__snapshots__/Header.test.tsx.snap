// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should render properly 1`] = `
<div>
  <div
    class="_header_63c2ea"
  >
    <h1>
      <span
        class="_icon_63c2ea _left_63c2ea"
        style="background-image: url(undefined);"
      />
    </h1>
  </div>
</div>
`;

exports[`Header > should render with iconPosition right 1`] = `
<div>
  <div
    class="_header_63c2ea"
  >
    <h1>
      <span
        class="_icon_63c2ea _right_63c2ea"
        style="background-image: url(undefined);"
      />
    </h1>
  </div>
</div>
`;

exports[`Header > should render with imgLogo 1`] = `
<div>
  <div
    class="_header_63c2ea"
  >
    <h1>
      <span
        class="_icon_63c2ea _left_63c2ea"
        style="background-image: url(mock-imgLogo);"
      />
    </h1>
  </div>
</div>
`;

exports[`Header > should render with title 1`] = `
<div>
  <div
    class="_header_63c2ea"
  >
    <h1>
      <span
        class="_icon_63c2ea _left_63c2ea"
        style="background-image: url(undefined);"
      />
      mock title
    </h1>
  </div>
</div>
`;
